log_downloader.py

```diff
@@ -7,6 +8,22 @@
 import wmi
 
 
+def open_latest_log_in_advantage_scope():
+    try:
+        log_files = [(f, f.stat().st_mtime) for f in LOGS_DIR.glob("*.wpilog")]
+
+        if not log_files:
+            logger.warning("No log files found")
+            return
+
+        latest_log = max(log_files, key=lambda x: x[1])[0]
+
+        logger.info(f"Opening {latest_log.name} with AdvantageScope")
+        subprocess.Popen([ADVANTAGE_SCOPE_PATH, str(latest_log)])
```

Comment:
From `subprocess` docs ([source](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module)):
> The recommended approach to invoking subprocesses is to use the [`run()`](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle. For more advanced use cases, the underlying [`Popen`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) interface can be used directly.

Also, you can simply open the file, without specifying AdvantageScope, by taking advantage of the file association Windows has between AdvnatageScope and `.wpilog` files.

Reply:
subprocess.run() would wait for the application to close before continuing.

Reply:
OK, good to know. And the second thing?

Reply:
The file association sometimes needs to be configured manually. Also the file association is for the wpilib advantage scope meaning if someone is using a beta version he will need to configure it manually.

Reply:
This should be done anyway, as it speeds up opening the file. Versions aren't released even weekly so beta version are not a problem.
If this script is a way to force you to make your life easier, than I'm all for it.

Reply:
3a2e2bf790c38d65aea860b8c4cf04a4a521b0dc